### DISCUSSION TITLE
SQLServer P4001 The introspected database was empty

### DISCUSSION THREAD

**Discussion by BobFitzsimons** [prisma/prisma#13304] [Q&A]
I have a SQL Server  test database test1, which has some tables.
I have TCP enabled on port 1433

My connection string is
DATABASE_URL="sqlserver://127.0.0.1:1433;initialCatalog=test1;integratedSecurity=true;trustServerCertificate=true"

I seem to get a connection ok but when I do a npx prisma db pull I get the following 

P4001 The introspected database was empty

Any help/ideas would be greatly appreciated.

**Comment by janpio** [comment]
This indicates that the database it is talking to is either indeed empty, or the tables are unexpected and can not be understood.
If you suspect the second option to be the case, please share the DDL SQL with us so we can reproduce.

**Comment by BobFitzsimons** [comment]
Hi
Hopefully this is the DDL to recreate the database. I am not using the default database file locations
Thanks
Bob

```
USE [master]
GO

/****** Object:  Database [test1]    Script Date: 13/05/2022 13:56:08 ******/
CREATE DATABASE [test2]
CONTAINMENT = NONE
ON  PRIMARY
( NAME = N'test2', FILENAME = N'D:\DB_Data\MSSQL\Data\test2.mdf' , SIZE = 8192KB , MAXSIZE = UNLIMITED, FILEGROWTH = 65536KB )
LOG ON
( NAME = N'test2_log', FILENAME = N'D:\DB_Data\MSSQL\Data\test2_log.ldf' , SIZE = 8192KB , MAXSIZE = 2048GB , FILEGROWTH = 65536KB )
WITH CATALOG_COLLATION = DATABASE_DEFAULT
GO

IF (1 = FULLTEXTSERVICEPROPERTY('IsFullTextInstalled'))
begin
EXEC [test1].[dbo].[sp_fulltext_database] @action = 'enable'
end
GO

ALTER DATABASE [test2] SET ANSI_NULL_DEFAULT OFF
GO

ALTER DATABASE [test2] SET ANSI_NULLS OFF
GO

ALTER DATABASE [test2] SET ANSI_PADDING OFF
GO

ALTER DATABASE [test2] SET ANSI_WARNINGS OFF
GO

ALTER DATABASE [test2] SET ARITHABORT OFF
GO

ALTER DATABASE [test2] SET AUTO_CLOSE OFF
GO

ALTER DATABASE [test2] SET AUTO_SHRINK OFF
GO

ALTER DATABASE [test2] SET AUTO_UPDATE_STATISTICS ON
GO

ALTER DATABASE [test2] SET CURSOR_CLOSE_ON_COMMIT OFF
GO

ALTER DATABASE [test2] SET CURSOR_DEFAULT  GLOBAL
GO

ALTER DATABASE [test2] SET CONCAT_NULL_YIELDS_NULL OFF
GO

ALTER DATABASE [test2] SET NUMERIC_ROUNDABORT OFF
GO

ALTER DATABASE [test2] SET QUOTED_IDENTIFIER OFF
GO

ALTER DATABASE [test2] SET RECURSIVE_TRIGGERS OFF
GO

ALTER DATABASE [test2] SET  DISABLE_BROKER
GO

ALTER DATABASE [test2] SET AUTO_UPDATE_STATISTICS_ASYNC OFF
GO

ALTER DATABASE [test2] SET DATE_CORRELATION_OPTIMIZATION OFF
GO

ALTER DATABASE [test2] SET TRUSTWORTHY OFF
GO

ALTER DATABASE [test2] SET ALLOW_SNAPSHOT_ISOLATION OFF
GO

ALTER DATABASE [test2] SET PARAMETERIZATION SIMPLE
GO

ALTER DATABASE [test2] SET READ_COMMITTED_SNAPSHOT OFF
GO

ALTER DATABASE [test2] SET HONOR_BROKER_PRIORITY OFF
GO

ALTER DATABASE [test2] SET RECOVERY FULL
GO

ALTER DATABASE [test2] SET  MULTI_USER
GO

ALTER DATABASE [test2] SET PAGE_VERIFY CHECKSUM
GO

ALTER DATABASE [test2] SET DB_CHAINING OFF
GO

ALTER DATABASE [test2] SET FILESTREAM( NON_TRANSACTED_ACCESS = OFF )
GO

ALTER DATABASE [test2] SET TARGET_RECOVERY_TIME = 60 SECONDS
GO

ALTER DATABASE [test2] SET DELAYED_DURABILITY = DISABLED
GO

ALTER DATABASE [test2] SET ACCELERATED_DATABASE_RECOVERY = OFF
GO

ALTER DATABASE [test2] SET QUERY_STORE = OFF
GO

ALTER DATABASE [test2] SET  READ_WRITE
GO

USE [test2]
GO

/****** Object:  Table [dbo].[Table_1]    Script Date: 13/05/2022 14:03:54 ******/
SET ANSI_NULLS ON
GO

SET QUOTED_IDENTIFIER ON
GO

CREATE TABLE [dbo].[Table_1](
       [x] [nvarchar](50) NULL
) ON [PRIMARY]
GO
```

**Reply by pantharshit00** [reply]
Hello @BobFitzsimons , our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

I wasn't able to reproduce this with 4.7.0. Can you please try again if this is still relevant?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
